Source/Diagnostics/ComputeDiagFunctors/BackTransformFunctor.cpp

```diff
@@ -202,6 +228,64 @@ BackTransformFunctor::LorentzTransformZ (amrex::MultiFab& data, amrex::Real gamm
         amrex::Array4< amrex::Real > arr = data[mfi].array();
         amrex::Real clight = PhysConst::c;
         amrex::Real inv_clight = 1.0_rt/clight;
+#ifdef WARPX_DIM_RZ
+        const int n_rz = WarpX::n_rz_azimuthal_modes;
+        const int n_rcomps = WarpX::ncomps;
+        amrex::ParallelFor( tbx,
+            [=] AMREX_GPU_DEVICE (int i, int j, int k)
+            {
+            // arr(x,y,z,comp) has (2*num_rz_modes-1)*10 components
+            // If there are 2 modes, then Er_0_real, Er_1_real, Er_2_real are the first three components
+            // followed by three components each for Et, Ez, Br, Bt, Bz, jr, jt, jz, and rho
+                for (int imode = 0; imode < n_rz; ++imode) {
+                    int offset = 1;
+                    // number of components for a given mode
+                    int ncomp = 2;
+                    if (imode == 0) {
+                        offset = 0;
+                        // Mode 0 has only real part, thus ncomp = 1
+                        ncomp = 1;
+                    }
+                    for (int icomp = 0; icomp < ncomp; ++icomp) {
```

Comment:
Could we replace the whole code above by a loop over components?
```suggestion
                for (int mode_comp_offset = 0; mode_comp_offset < nr_comp; ++mode_comp_offset) {
```